server/src/WebSocketEvents.js

```diff
@@ -290,20 +290,89 @@ eventResponses.mv_r = (clientSocket) => {
  * @param {*} clientSocket
  * @param {String} data
  */
+
+// TODO: Not sure where to put this
+let entityChatCooldownList = [
+    // { id: 1, scope: "GLOBAL" },
+];
+
 eventResponses.chat = (clientSocket, data) => {
-    if (!data) return;
+    // Can't use ChatState here 'cause it's located outside module
+    const CHAT_SCOPES = {
+        LOCAL: {
+            value: "LOCAL",
+            cooldown: 0,
+        },
+        GLOBAL: {
+            value: "GLOBAL",
+            cooldown: 3000,
+        },
+        TRADE: {
+            value: "TRADE",
+            cooldown: 3000,
+        },
+    };
+
+    if (!data || !data.scope || !data.message) return;
+    if (data.message.length > 255) return;
     if (clientSocket.inGame === false) return;
 
+    const { message, scope } = data;
     const { entity } = clientSocket;
 
     // Ignore this event if they are dead.
     if (entity.hitPoints <= 0) return;
 
+    const { cooldown } = Object
+        .values(CHAT_SCOPES)
+        .find((_scope) => _scope.value === scope);
```

Comment:
Can just do `CHAT_SCOPES[scope]` if the property keys are always the scope names.